client/solve.go

```diff
@@ -109,7 +109,9 @@ func (c *Client) solve(ctx context.Context, def *llb.Definition, runGateway runG
 		}
 	}
 
-	cacheOpt, err := parseCacheOptions(opt)
+	statusContext = bklog.WithLogger(statusContext, bklog.GetLogger(statusContext).WithField("session", s.ID()))
+
+	cacheOpt, err := parseCacheOptions(statusContext, opt)
```

Comment:
why pass statusContext in here instead of `ctx`. parsing doesn't seem to be related to session

Reply:
I think they are in same scope `resolve`, should with same logger and same tracing.

Reply:
I don't understand. In fact, I think it would be more correct if this `parse` func was called before the session is even created.

Reply:
I just think in tracing view.

The flow in `solve`.

```
Solve:

NewSection(ctx) ->  parseCache(ctx) -> doOthers(ctx) 
```

I feel the ctx should be the same one.

Reply:
I revert this.
this change may out of the PR scope.